examples/reader-writer.cc

```diff
@@ -0,0 +1,426 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <cassert>
+#include <fstream>
+#include <iostream>
+#include <list>
+#include <memory>
+
+#include <parquet/api/writer.h>
+#include <parquet/api/reader.h>
+
+/*
+ * This example describes writing and reading Parquet Files in C++ and serves as a reference to the API.
+ * The file contains all the physical data types supported by Parquet.
+**/
+
+/* Parquet is a structured columnar file format
+ * Parquet File = "Parquet data" + "Parquet Metadata"
+ * "Parquet data" is simply a vector of RowGroups. Each RowGroup is a batch of rows in a columnar layout
+ * "Parquet Metadata" contains the "file schema" and attributes of the RowGroups and their Columns
+ * "file schema" is a tree where each node is either a primitive type (leaf nodes) or a complex (nested) type (internal nodes)
+ * For specific details, please refer the format here:
+ * https://github.com/apache/parquet-format/blob/master/LogicalTypes.md
+**/
+
+constexpr int NUM_ROW_GROUPS = 10;
+constexpr int NUM_ROWS_PER_ROW_GROUP = 512;
+constexpr int NUM_COLUMNS = 8;
+constexpr int FIXED_LENGTH = 10;
+const std::string PARQUET_FILENAME = "parquet_cpp_example.parquet";
+
+class ParquetWriter {
+ public:
+  ParquetWriter() {
+      ostream_.reset(new parquet::InMemoryOutputStream());
+  }
+
+  void SetupSchema() {
+    parquet::schema::NodeVector fields;
+    // Create a primitive node named 'boolean_field' with type:BOOLEAN, repetition:REQUIRED
+    fields.push_back(parquet::schema::PrimitiveNode::Make("boolean_field", parquet::Repetition::REQUIRED, parquet::Type::BOOLEAN, parquet::LogicalType::NONE));
```

Comment:
At first glance, I'm ok with with not doing `using parquet::*` but this line gets really long because of that. I would prefer to have less `parquet::` prefixes here.